src/Storage/Storage/Common/Util.cs

```diff
@@ -855,5 +855,20 @@ public static Uri RemoveFileUriTrailingDot(Uri fileUri)
             }
             return new Uri(noLastSegment);
         }
+
+        public static string RemoveFilePathTrailingDot(string FilePath)
+        {
+            string filePathWithoutDot;
+            string[] segments = FilePath.Split(NamingUtil.PathSeparators);
+            List<string> segmentsWithoutDot = new List<string>();
+
+            foreach (string segment in segments)
+            {
+                segmentsWithoutDot.Add(segment.TrimEnd('.'));
+            }
+            filePathWithoutDot = String.Join("/", segmentsWithoutDot.ToArray());
```

Comment:
1. `Path.DirectorySeparatorChar` is recommended here for cross-platform compatibility
2. `ToArray()` is not necessary since List inherits IEnumerable

Reply:
Done